float32 tau
float32 spd
float32 pos
